.github/workflows/pr_verification.yml

```diff
@@ -176,3 +177,129 @@ jobs:
             -Dsonar.pullrequest.key=${{ github.event.pull_request.number }}
             -Dsonar.pullrequest.base=${{ github.base_ref }}
             -Dsonar.pullrequest.branch=${{ github.head_ref }}
+
+  build_image:
+    name: build Candlepin image
+    runs-on: ubuntu-latest
+    permissions:
+      contents: read
+      packages: write
+
+    steps:
+      - name: Install packages
+        run: |
+          sudo apt-get update
+          sudo apt-get -y install \
+            git-core \
+            buildah
+
+      - name: Check out repository
+        uses: actions/checkout@v3
+
+      - name: Log in to the registry
+        uses: docker/login-action@v2
+        with:
+          registry: ghcr.io
+          username: ${{ github.actor }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Build Candlepin image
+        id: build-image
+        uses: redhat-actions/buildah-build@v2
+        with:
+          image: ghcr.io/candlepin/gha_build
+          tags: pr-${{ github.event.pull_request.number }}
+          containerfiles: ./.github/containers/Dockerfile
+
+      - name: Push to registry
+        uses: redhat-actions/push-to-registry@v2
+        with:
+          image: ghcr.io/candlepin/gha_build
+          tags: pr-${{ github.event.pull_request.number }}
+          registry: ghcr.io
+          username: ${{ github.actor }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+
+  spec_tests:
+    name: spec tests
+    runs-on: ubuntu-latest
+    container:
+      image: fedora:38
+      options: --privileged
+    permissions:
+      contents: read
+      packages: write
+    needs: build_image
+    strategy:
+      fail-fast: false
+      matrix:
+        database: [postgres, mariadb]
+        mode: [standalone, hosted]
+
+    steps:
+      - name: Check out repository
+        uses: actions/checkout@v3
+
+      - name: Install dependencies
+        shell: bash
+        run: |
+          dnf -y --setopt install_weak_deps=False update
+          dnf --setopt install_weak_deps=False install -y gettext jss docker
+
+      - name: Install docker compose
+        shell: bash
+        run: |
+          DOCKER_CONFIG=${DOCKER_CONFIG:-$HOME/.docker}
+          mkdir -p $DOCKER_CONFIG/cli-plugins
+          curl -SL https://github.com/docker/compose/releases/download/v2.18.1/docker-compose-linux-x86_64 -o $DOCKER_CONFIG/cli-plugins/docker-compose
+          sudo chmod +x $DOCKER_CONFIG/cli-plugins/docker-compose
```

Comment:
We need to install docker compose v2 so that we can use the `--wait` flag when creating the DB and Candlepin container. Unfortunately I wasn't able to download this packages using dnf and so I resorted to downloading and installing docker compose v2 using this step.